workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectDispatcherSystem.cs

```diff
@@ -12,26 +12,26 @@ namespace Improbable.Gdk.Core.GameObjectRepresentation
     [UpdateInGroup(typeof(SpatialOSReceiveGroup.GameObjectReceiveGroup))]
     internal class GameObjectDispatcherSystem : ComponentSystem
     {
-        private readonly Dictionary<int, MonoBehaviourActivationManager> entityIndexToActivationManager =
-            new Dictionary<int, MonoBehaviourActivationManager>();
-        private readonly Dictionary<int, InjectableStore> entityIndexToReaderWriterStore =
-            new Dictionary<int, InjectableStore>();
+        private readonly Dictionary<Entity, MonoBehaviourActivationManager> entityToActivationManager =
+            new Dictionary<Entity, MonoBehaviourActivationManager>();
+        private readonly Dictionary<Entity, InjectableStore> entityToReaderWriterStore =
+            new Dictionary<Entity, InjectableStore>();
 
         public readonly List<GameObjectComponentDispatcherBase> GameObjectComponentDispatchers =
             new List<GameObjectComponentDispatcherBase>();
 
         private RequiredFieldInjector injector;
         private ILogDispatcher logger;
 
-        internal void RemoveActivationManagerAndReaderWriterStore(int entityIndex)
+        internal void RemoveActivationManagerAndReaderWriterStore(Entity entity)
         {
-            if (!entityIndexToActivationManager.ContainsKey(entityIndex))
+            if (!entityToActivationManager.ContainsKey(entity))
             {
-                throw new ActivationManagerNotFoundException($"MonoBehaviourActivationManager not found for entityIndex {entityIndex}.");
+                throw new ActivationManagerNotFoundException($"MonoBehaviourActivationManager not found for entity {entity.Index}.");
```

Comment:
Nit: Could we print the `entity.Version` here too? Not sure if that's ever going to be useful. Just making sure we stay safe.